Sources/Common/ProgressiveRouteType.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  ProgressiveRouteType.swift
+//  OperaSwift
+//
+//  Created by Diego Ernst on 5/11/17.
+//
+//
+
+import Alamofire
+import Foundation
+
+public protocol ProgressiveRouteType: AdaptedRouteType {
```

Comment:
So `AdaptedRouteType` is a generic wrapper to RouteType, it would allow developers to have a type with a nested routetype and make it a new routetype out of the box.